### DISCUSSION TITLE
Displaying Spatie Media Collection in Info List

### DISCUSSION THREAD

**Discussion by sherifiskander** [filamentphp/filament#9068] [Help]
Got a model that has multiple images attached to it. How can I make a repeater for an image entry out of a collection. The idea is to display the images individually since I’m attaching a download action to each.

**Comment by sinnbeck** [comment]
In infolists its called a RepeatableEntry. Pass the relation name to make() and I assume it works (I have only used it for hasMany)

https://filamentphp.com/docs/3.x/infolists/entries/repeatable#overview

I dont think they support action but you can add the `->url()` method to download the image perhaps :)

**Reply by sherifiskander** [reply]
I tried that. The Spatie Media library returns a collection. They all get bundled together as one image. I got the action working, but now in field with multiple images I can download them all in a zip file. Can’t download the individual ones.

**Reply by sinnbeck** [reply]
Can you show the code that "kinda" works ?

**Reply by sherifiskander** [reply]
SpatieMediaLibraryImageEntry::make('Other Imagery')->collection('Additonal Images')->conversion('thumb')
                ->Action(
                Action::make('downloadImage')->label('Download Image')
                    ->action(function ($record) {
                        $mediaItem = $record->getMedia("Additonal Images");
                        return MediaStream::create('images.zip')->addMedia($mediaItem);
                        ;
                    })
                )

**Reply by sherifiskander** [reply]
This displays the images in the collection side by side and allows the download as a zip file. Tried putting the collection inside a repeatable entry but no go

**Reply by sinnbeck** [reply]
Did you try adding `->url()` instead of an action ?

```php
->url(fn($record) => $record->getImageUrl(), true)

**Reply by sherifiskander** [reply]
getImageUrl() is not available for Spatie Media Library records..

url(fn($record) => $record->getFirstMediaUrl(), true) works, but this just opens the image in a new tab, not download it

**Reply by sinnbeck** [reply]
You can perhaps call a controller that gets the id of the record instead and forces a download of the file.

https://spatie.be/docs/laravel-medialibrary/v10/downloading-media/downloading-a-single-file

**Reply by sherifiskander** [reply]
This is exactly what I'm doing with the action for the single images.  My issue is when I don't use ->first(), the function returns the collection which I want to have listed as individual images in a repeatable entry so I can display the individual images and create a download action for them

action(function ($record) {
                        $mediaItem = $record->getMedia("Image")->first();
                        return $mediaItem;
                        ;
                    })

**Reply by LyndonMcjohnson** [reply]
hello @sherifiskander were you able to figure this out eventually? If yes, can you please share your approach? Thanks!
